src/ogd/games/BLOOM/features/PerCountyFeature.py

```diff
@@ -0,0 +1,31 @@
+# import libraries
+import logging
+from typing import Optional, List
+# import locals
+from ogd.core.utils.Logger import Logger
+from ogd.core.generators.Generator import GeneratorParameters
+from ogd.core.generators.extractors.PerCountFeature import PerCountFeature
+from ogd.core.models.Event import Event
+
+class PerCountyFeature(PerCountFeature):
+    COUNTY_LIST = ["Hillside", "Forest", "Prairie", "Wetland", "Urban"]
+
+    def __init__(self, params: GeneratorParameters):
+        super().__init__(params=params)
+
+    # *** IMPLEMENT ABSTRACT FUNCTIONS ***
+
+    def _validateEventCountIndex(self, event: Event):
+        ret_val: bool = False
+
+        county_name = event.GameState.get('county_name', event.EventData.get('county_name', "COUNTY NAME NOT FOUND"))
+        if county_name is not None:
```

Comment:
One minor tweak: `county_name` is never `None`, because we gave it a default value in the call to `.get(...)`, namely `"COUNTY NAME NOT FOUND"`.

So, instead of checking that `county_name` is not `None`, check that it's not `"COUNTY NAME NOT FOUND"`.

Reply:
Alternately, you could just remove the default value, since it doesn't actually get used. So `county_name = event.GameState.get('county_name', event.EventData.get('county_name'))`